include/sysdep.h

```diff
@@ -211,15 +211,15 @@ typedef uint_least16_t uint16;
 
 /* find out operating system if not specified on the command line */
 
-#if defined(_WIN32) || defined(__WIN32__)
-#  ifndef WIN32
-#    define WIN32 1
+#if defined(__WIN32__)
+#  ifndef _WIN32
+#    define _WIN32 1
```

Comment:
Hmm I'm not sure we should do this. Manually defining what is a well known symbol is asking for trouble IMO. 

Having said that, `WIN32` is defined by CMake. I think we should just be leveraging CMake completely for a lot of this system defintion stuff. https://cmake.org/cmake/help/latest/variable/WIN32.html

Reply:
Thought about it a bit and I do think it makes more sense to use native flags instead of cmake ones. Not everyone uses cmake, and so we want to reduce the amount of compiler definitions necessary to use our headers as much as possible I think.

Reply:
How do you mean? What is the alternative to CMake here? That is the build system that Csound uses. So there is no other alternative. It's a mistake to over generalise on things that are definite. We use Cmake, we should leverage it as much as possible so long as that is the case.

Reply:
No, what I mean is: suppose someone wants to use write a C program and link to libcsound. But suppose that someone isn't using cmake. Then they can't use our exported targets. So, to use our headers, they'll have to define all the config variables by hand first.

Reply:
Ah right, sysdep is client facing. In that case we have to account for each supported compiler/platform as best as possible. I guess it is doing that already. Best we can do is to update it for latest GCC/Clang/MSVC versions.